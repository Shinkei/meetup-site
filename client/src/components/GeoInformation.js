import React, {Component} from 'react'
import loadGoogleMapsAPI from 'load-google-maps-api'

const API_CONFIG = {
  key: 'AIzaSyAuBEdfk-djsMYt8nSkNHUPplqrokZ4UPY',
  language: 'es'
}

const MAP_STYLES = {
  height: '450px',
  width: '100%'
}

class GeoInformation extends Component {
  componentDidMount () {
    
    const center= {
      lat:this.props.latitude,
      lng:this.props.longitud
    };

    const options = {
      center,
      zoom: this.props.zoom    
    }

    loadGoogleMapsAPI(API_CONFIG).then(googleMaps => {
      const map = new googleMaps.Map(this.refs.map, options)
      new googleMaps.Marker({position:center, map})
    }).catch(err => {
      console.warning('Something went wrong loading the map', err)
    })
  }

  render () {
    return (
      <article>
        <h3>Ubicacion</h3>
        <h4>Direcci√≥n</h4>
        <p>{this.props.direccion}</p>
        <div ref='map' style={MAP_STYLES} />
      </article>
    )
  }
}

export default GeoInformation
