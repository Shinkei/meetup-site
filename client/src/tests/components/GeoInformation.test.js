import React from 'react'
import renderer from 'react-test-renderer'
import GeoInformation from '../../components/GeoInformation'

describe('GeoInformation Component', () => {
  it('should not regress', () => {
    const tree = renderer.create(<GeoInformation />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
