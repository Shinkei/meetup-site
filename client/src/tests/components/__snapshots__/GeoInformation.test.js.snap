// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoInformation Component should not regress 1`] = `
<article>
  <h3>
    Ubicacion
  </h3>
  <h4>
    Direcci√≥n
  </h4>
  <p />
  <div
    style={
      Object {
        "height": "450px",
        "width": "100%",
      }
    }
  />
</article>
`;
